specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-01-01/frontdoor.json

```diff
@@ -0,0 +1,2292 @@
+      "allOf": [
+        {
+          "$ref": "#/definitions/RoutingRuleUpdateParameters"
+        }
+      ],
+      "properties": {
+        "resourceState": {
+          "description": "Resource status.",
+          "$ref": "#/definitions/ResourceState"
+        }
+      }
+    },
+    "RoutingRuleListResult": {
+      "description": "Result of the request to list Routing Rules. It contains a list of Routing Rule objects and a URL link to get the next set of results.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/RoutingRule"
+          },
+          "description": "List of Routing Rules within a Front Door."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "URL to get the next set of RoutingRule objects if there are any."
+        }
+      }
+    },
+    "RoutingRuleUpdateParameters": {
+      "description": "Routing rules to apply to an endpoint",
+      "type": "object",
+      "properties": {
+        "frontendEndpoints": {
+          "description": "Frontend endpoints associated with this rule",
+          "type": "array",
+          "items": {
+            "description": "A reference to a frontend endpoint",
+            "$ref": "./network.json#/definitions/SubResource"
+          }
+        },
+        "acceptedProtocols": {
+          "description": "Protocol schemes to match for this rule",
+          "type": "array",
+          "items": {
+            "type": "string",
+            "description": "Accepted protocol schemes.",
+            "enum": [
+              "Http",
+              "Https"
+            ],
+            "x-ms-enum": {
+              "name": "FrontDoorProtocol",
+              "modelAsString": true
+            }
+          }
+        },
+        "patternsToMatch": {
+          "description": "The route patterns of the rule.",
+          "type": "array",
+          "items": {
+            "type": "string",
+            "description": "A route pattern of the rule. Must not have any * except possibly after the final / at the end of the path."
+          }
+        },
+        "enabledState": {
+          "description": "Whether to enable use of this rule. Permitted values are 'Enabled' or 'Disabled'",
+          "enum": [
+            "Enabled",
+            "Disabled"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "routingRuleEnabledState",
+            "modelAsString": true
+          }
+        },
+        "routeConfiguration": {
+          "description": "A reference to the routing configuration.",
+          "$ref": "#/definitions/RouteConfiguration"
+        },
+        "rulesEngine": {
+          "description": "A reference to a specific Rules Engine Configuration to apply to this route.",
+          "$ref": "./network.json#/definitions/SubResource"
+        }
```

Comment:
this is a breaking change for older SDK/CLI/PS.. clients.. . what hapens if they send request without this property ? (but resource has it)..

Reply:
same here.. breaking change guide : [https://microsoft.sharepoint.com/:w:/t/azureresourcemanagerteam/EWXsAQ1yx25KkyYCeeWGUgwBSBxEdUDEbHi6FZ__U8EOQw?e=xEcppo](https://microsoft.sharepoint.com/:w:/t/azureresourcemanagerteam/EWXsAQ1yx25KkyYCeeWGUgwBSBxEdUDEbHi6FZ__U8EOQw?e=xEcppo)

New property added to response 

If a new property/field is added to the response an API, the GET-PUT pipeline will be broken. Consider the case where from portal a customer updates the value of a new property "A". Another customer does a GET of this resource using the SDK. The SDK will ignore the property since it does not understand it. From the SDK, the customer does a PUT using the model that was returned from the GET. This will overwrite the change made by the first customer from the portal.  

---
In reply to: [367709076](https://github.com/Azure/azure-rest-api-specs/pull/8115#discussion_r367709076) [](ancestors = 367709076)

Reply:
Yes, we are aware that this is a breaking change and that's why we are updating the version as per the below guideline. 

Breaking Changes  

At a high level, changes to the contract of an API constitute a breaking change. Changes that impact backwards compatibility of an API is also considered a breaking change. **Teams MAY define backwards compatibility as their business needs require. For example, Azure defines the addition of a new JSON field in a response to be not backwards compatible.** Anything that would violate the Principle of Least Astonishment is considered a breaking change in Azure. Below are some concrete examples of what constitutes a breaking change. **In the below breaking change scenarios, the API version must be changed.**

Reply:
yeah.. this is new api version. my bad.. looks good.

---
In reply to: [367714311](https://github.com/Azure/azure-rest-api-specs/pull/8115#discussion_r367714311) [](ancestors = 367714311)